virtcontainers/network.go

```diff
@@ -1339,6 +1177,93 @@ func createVirtualNetworkEndpoint(idx int, ifName string, interworkingModel NetI
 	return endpoint, nil
 }
 
+func generateInterfacesAndRoutes(networkNS NetworkNamespace) ([]*grpc.Interface, []*grpc.Route, error) {
```

Comment:
I'm pretty surprised the linters don't complain about the length/complexity of this function ;)